### PR TITLE
Kj diettracker03

### PR DESCRIPTION
**Pull request by ghost**
- modified BasicFood Class: added protein, fat, carbs, equals() & hashcode(). made it immutable
- added Portion Class (BasicFood with portion size)
- added interfaces 
  - FoodService (for query from UI to Logic)
  - FoodService_Persistance(query from Logic to Persistence)
- added FoodServiceFactory. Static factory to provide instances of FoodService  interface
- tests for all

Hope it makes sense

### CODE REVIEW

**Comment by rgoolishian** [comment]
Looks good to me Kurt.

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  <artifactId>DietTracker</artifactId>
+  <packaging>jar</packaging>
+  <version>1.0-SNAPSHOT</version>
+  <name>DietTracker</name>
+  <url>http://maven.apache.org</url>
+  <dependencies>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>3.8.1</version>
```
Comment (smarks) [inline pom.xml:14]: use Junit 4.x not 3

File: src/main/java/edu/uml/diet/BasicFood.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        BasicFood basicFood = (BasicFood) o;
+
+        if (calories != basicFood.calories) return false;
```
Comment (smarks) [inline src/main/java/edu/uml/diet/BasicFood.java:71]: always use braces around if statements

File: src/test/java/edu/uml/diet/BasicFoodTest.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+ */
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+
```
Comment (smarks) [inline src/test/java/edu/uml/diet/BasicFoodTest.java:14]: don't forget class comments

File: src/test/java/edu/uml/diet/BasicFoodTest.java [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    public void testGetCaloriesNegative(){
+        BasicFood basicfood = new BasicFood(foodName, calories, fat, carbs, protein);
+        assertFalse("testGetCalories: negative", (calories+10) == basicfood.getCalories());
+    }
+
+    /**
+     * test to verify that correct  fat  is returned by getFat()
+     */
+    @Test
+    public void testGetFat(){
```
Comment (smarks) [inline src/test/java/edu/uml/diet/BasicFoodTest.java:71]: Don't forget D.R.Y. - you make basicFood a member variable and initialize it once in a setup method.

**Comment by smarks** [comment]
looks good.

**Pull request merged**